test/uk/gov/hmrc/merchandiseinbaggagefrontend/BaseSpec.scala

```diff
@@ -38,13 +38,26 @@ trait BaseSpecWithApplication extends BaseSpec
   lazy val injector: Injector = app.injector
 
   lazy val declarationJourneyRepository: DeclarationJourneyRepository = app.injector.instanceOf[DeclarationJourneyRepository]
-}
 
+  override def beforeEach(): Unit = {
+    declarationJourneyRepository.deleteAll().futureValue
+  }
 
-trait BaseSpecWithWireMock extends BaseSpecWithApplication {
-  val paymentMockServer = new WireMockServer(9057)
+}
 
-  override def beforeEach: Unit = paymentMockServer.start()
 
-  override def afterEach: Unit = paymentMockServer.stop()
+trait BaseSpecWithWireMock extends BaseSpecWithApplication with BeforeAndAfterEach {
+  val paymentMockServer = new WireMockServer(9057)
```

Comment:
We might just have a single server instance with an arbitrary port and override the config in BaseSpecWithApplication to point all 'connectors' at it? 

Thats less code duplication when we add more connectors and also means you can run the services locally via service manager without the tests failing.

Reply:
Like this:
https://github.com/hmrc/tps-payments-frontend/blob/master/test/support/WireMockSupport.scala

https://github.com/hmrc/tps-payments-frontend/blob/e1b82dbee816ac618aefec9627eb73e23412f61c/test/support/ItSpec.scala#L89

Reply:
good idea

Reply:
Couldn't get this to work, will revisit later.